micropython/utarfile-write/utarfile/write.py

```diff
@@ -0,0 +1,135 @@
+"""Additions to the TarFile class to support creating and appending tar files.
+
+The methods defined below in are injected into the TarFile class in the
+utarfile package.
+"""
+
+import uctypes
+import os
+
+# Extended subset of tar header fields including the ones we'll write.
+# http://www.gnu.org/software/tar/manual/html_node/Standard.html
+_TAR_HEADER = {
+    "name": (uctypes.ARRAY | 0, uctypes.UINT8 | 100),
+    "mode": (uctypes.ARRAY | 100, uctypes.UINT8 | 7),
+    "uid": (uctypes.ARRAY | 108, uctypes.UINT8 | 7),
+    "gid": (uctypes.ARRAY | 116, uctypes.UINT8 | 7),
+    "size": (uctypes.ARRAY | 124, uctypes.UINT8 | 12),
+    "mtime": (uctypes.ARRAY | 136, uctypes.UINT8 | 12),
+    "chksum": (uctypes.ARRAY | 148, uctypes.UINT8 | 8),
+    "typeflag": (uctypes.ARRAY | 156, uctypes.UINT8 | 1),
+}
+
+# Following https://github.com/python/cpython/blob/3.11/Lib/tarfile.py
```

Comment:
Does the code here copy anything verbatim from that Python file?  Just need to be careful of licensing.